crates/sparrow-instructions/src/inst.rs

```diff
@@ -236,15 +232,43 @@ pub enum InstKind {
     ///
     /// The number of arguments should match the number of fields.
     Record,
+    /// A user defined function.
+    Udf(Arc<dyn Udf>),
+}
+
+impl Clone for InstKind {
+    fn clone(&self) -> Self {
+        match self {
+            Self::Simple(arg0) => Self::Simple(arg0.clone()),
+            Self::FieldRef => Self::FieldRef,
+            Self::Cast(arg0) => Self::Cast(arg0.clone()),
+            Self::Record => Self::Record,
+            Self::Udf(arg0) => Self::Udf(arg0.clone()),
+        }
+    }
+}
+
+impl PartialEq for InstKind {
```

Comment:
I can't derive `PartialEq` -- something about it moving `self` out. Can look at this together if you think we can get around that.

Reply:
that's somewhat surprising -- I can go either way.